lib/cash-in/cash-in-tx.js

```diff
@@ -133,20 +133,18 @@ function postProcess (r, pi, isBlacklisted, addressReuse) {
       }
     })
     .then(sendRec => {
-      settingsLoader.loadLatest().then(it => {
-        const rejectAddressReuseActive = configManager.getCompliance(it.config).rejectAddressReuse
-        if (rejectAddressReuseActive) {
-          blacklist.addToUsedAddresses(r.tx.toAddress, r.tx.cryptoCode)
-            .catch(err => logger.error('Failure adding to addressReuse', err))
-        }
-      })
-
       pi.notifyOperator(r.tx, sendRec)
         .catch((err) => logger.error('Failure sending transaction notification', err))
       return logAction(sendRec, r.tx)
     })
 }
 
+function getAddresses () {
+  const sql = `SELECT DISTINCT to_address FROM cash_in_txs`
+
+  return db.any(sql).then(res => res.map(it => it.to_address))
+}
+
```

Comment:
This will run on every tx update, take a look at how the `checkForBlacklisted` function works and do the same here. Also it might be better to return a boolean from the sql instead of getting this huge list to go through in js.